examples/controls_example/public/react_controls/data_controls/data_control_editor.tsx

```diff
@@ -283,8 +292,23 @@ export const DataControlEditor = <State extends DataControlEditorState = DataCon
                   dataView={selectedDataView}
                   onSelectField={(field) => {
                     setEditorState({ ...editorState, fieldName: field.name });
-                    setSelectedControlType(fieldRegistry?.[field.name]?.compatibleControlTypes[0]);
 
+                    /**
+                     * make sure that the new field is compatible with the selected control type and, if it's not,
+                     * reset the selected control type to the **first** compatible control type
+                     */
+                    const newCompatibleControlTypes =
+                      fieldRegistry?.[field.name]?.compatibleControlTypes ?? [];
+                    if (
+                      !selectedControlType ||
+                      !newCompatibleControlTypes.includes(selectedControlType!)
+                    ) {
+                      setSelectedControlType(newCompatibleControlTypes[0]);
+                    }
```

Comment:
This is just a super small UI improvement - previously, if you had range slider selected and chose a different number field, it would reset you back to an options list control. I personally found this a bit frustrating, so we are not defaulting to **keep** the previous control type selected, as long as it is still compatible.